operator/pkg/controller/virtualippool/controller.go

```diff
@@ -57,7 +75,177 @@ func (r *VirtualIPPoolReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return utils.Requeue(err)
 	}
 
+	// Retrieve all nodes
+	if err := r.client.List(ctx, &r.nodeList); err != nil {
+		return utils.Requeue(err)
+	}
+
+	// Retrieve all configured Virtual Router IDs
+	r.usedVRID = make(map[int]bool, 255)
+	pools := metalk8sscalitycomv1alpha1.VirtualIPPoolList{}
+	if err := r.client.List(ctx, &pools); err != nil {
+		return utils.Requeue(err)
+	}
+	for _, pool := range pools.Items {
+		if err := r.retrieveVRID(ctx, pool); err != nil {
+			return utils.Requeue(err)
+		}
+	}
+
+	// We consider all objects has "to be updated" then the `CreateOrUpdate` function
+	// will see (using the mutate function) if those object actually need updates or not
+	objsToUpdate := []client.Object{
+		r.instance.GetConfigMap(),
+	}
+
+	handler := utils.NewObjectHandler(&r.instance, r.client, r.scheme, reqLogger, componentName, appName)
+
+	changed, err := handler.CreateOrUpdateOrDelete(ctx, objsToUpdate, nil, r.mutate)
+	if err != nil {
+		return utils.Requeue(err)
+	}
+	if changed {
+		return utils.EndReconciliation()
+	}
+
 	// TODO(user): your logic here
 
 	return utils.EndReconciliation()
 }
+
+func (r *VirtualIPPoolReconciler) retrieveVRID(ctx context.Context, pool metalk8sscalitycomv1alpha1.VirtualIPPool) error {
```

Comment:
> IIUC, we need it as a consequence of the "magic create/update/delete with a `mutate` handler" pattern

Not really, we need it, because we do not want to use the same VRID as another VirtualIPPool VIP so we retrieve all the VRID from all the VirtualIPPools

I will rename it and add a comment

Reply:
Done (already squashed for this one)

Reply:
Yes that part I got, but the fact that we need it attached to the reconciler as a struct field is only because of the `mutate`, right? Because we refresh this cache on every `Reconcile` call, no? If we don't, then we may be missing invalidation (when we remove a pool / an address from a pool).

Reply:
Oh yes indeed, it could be just a variable in the reconcile function and then we add a "lambda" function so that we give this argument to the mutate, but it will be almost the same. And indeed it's re-computed at every reconcile loop